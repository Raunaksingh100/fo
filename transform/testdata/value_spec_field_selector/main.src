package main

type A[T] struct{
	x T
}

type B[T, U] struct{
	x T
	y U
}

func main() {
	var a0 A[string] = A[string]{
		x: "foo",
	}
	var _ string = a0.x
	a1 := A[string]{
		x: "foo",
	}
	print(a1)

	var b0 B[string, int] = B[string, int]{
		x: "bar",
		y: 42,
	}
	var _ string = b0.x
	var _ int = b0.y
	b1 := B[string, int]{
		x: "bar",
		y: 42,
	}
	print(b1)
}
