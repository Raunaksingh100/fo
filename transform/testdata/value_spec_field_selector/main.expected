package main

type A struct {
	x interface{}
}

type B struct {
	x interface{}
	y interface{}
}

func main() {
	var a0 A = A{
		x: "foo",
	}
	var _ string = a0.x.(string)
	a1 := A{
		x: "foo",
	}
	print(a1)

	var b0 B = B{
		x: "bar",
		y: 42,
	}
	var _ string = b0.x.(string)
	var _ int = b0.y.(int)
	b1 := B{
		x: "bar",
		y: 42,
	}
	print(b1)
}
